x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -670,8 +721,14 @@ export class TaskRunner<
       if (!event.message) {
         event.message = `${this.ruleType.id}:${ruleId}: execution failed`;
       }
+      monitoringHistory.success = false;
     }
 
+    ruleMonitoring.execution.history.push(monitoringHistory);
+    ruleMonitoring.execution.calculated_metrics.success_ratio =
+      ruleMonitoring.execution.history.filter(({ success }) => success).length /
```

Comment:
nit - Do we want to only calculate the success_ratio if we have at least 10 items in the history?

Reply:
Why 10?

Reply:
that's the first number who come in my mind? if you a better one and a logical one, go for it

Reply:
I'm worried that folks will know their rule ran a few times but then they won't see a success ratio in the UI. I'm not sure we even have to limit - the ratio is valuable for two executions, just as it is for 200 I'd think

Reply:
Yes, You are right